src/tools/compiletest/src/json.rs

```diff
@@ -72,7 +72,8 @@ fn parse_line(file_name: &str, line: &str) -> Vec<Error> {
                 expected_errors
             }
             Err(error) => {
-                panic!("failed to decode compiler output as json: `{}`", error);
+                panic!("failed to decode compiler output as json: `{}` when parsing: {}", error,
```

Comment:
yeah, I think I agree with @alexcrichton, seems like we should pass in `output` and -- in the event of error -- print both the entire output *and* the failed `line`. It might be better still to pass in more context (such as stderr etc) -- basically the `ProcRes` struct, I think it's called? Then we can just call `fatal_proc_res` which dumps a lot of context iirc.